game-engine/src/jvmTest/kotlin/za/co/entelect/challenge/game/delegate/renderer/WormsRendererTest.kt

```diff
@@ -205,4 +210,66 @@ class WormsRendererTest {
         assertFalse(rendererConsole.commandPrompt(wormsPlayer).toLowerCase().contains("not supported"))
     }
 
+    @Test
+    fun testNullPlayerJsonRender() {
+        val wormsPlayers = buildWormsPlayerDefault(config)
+//        val lastPlayer = wormsPlayers.last()
+
+        val wormsMapGenerator = WormsMapGenerator(config, 0)
+        val wormsMap = wormsMapGenerator.getMap(wormsPlayers)
+
+        val rendererJson = WormsRendererJson(config)
+        val jsonFileString = rendererJson.render(wormsMap, null)
+
+        val jsonPropertiesShouldNotExist = listOf("consecutiveDoNothingCount", "myPlayer")
+        val propertiesFound = jsonPropertiesShouldNotExist.filter { prop ->
+            val index = jsonFileString.lines().indexOfFirst { it.contains(prop) }
+            index > -1
+        }
+        assertTrue(propertiesFound.isEmpty(), "JSON state file(null perspective) should not have properties >> " +
+                "[" + propertiesFound.joinToString(separator = ", ") + "]")
+//        Caused by:
+//        java.lang.UnsupportedOperationException: Interface can't be instantiated! Interface name: za.co.entelect.challenge.game.engine.powerups.Powerup
+/*        val gameDetails = Gson().fromJson(jsonFileString, WormsGameDetails::class.java)
```

Comment:
Remove or uncomment

Reply:
Is there a better way of testing this part? the deserialization gives some problems

Reply:
Since our tests are only for the jvm, You can look at json assert or json path testing libraries

Reply:
Found a way to lightly check for changes on this. We really only care about how many opponents there are, can you see weapons or not, etc.